File: Trabalho_Final/TF_V1/Disciplina.py
class disciplina(object):
     
    disc = None
    vecAlu = []

    def __init__(self, disc):
        self.disc = disc

    def Matricular(self, aluno):
        if aluno not in self.vecAlu:
            self.vecAlu.append(aluno)
    
    def Desmatricular(self, disc):
        if disc in self.vecAlu:
            ind = self.checkInd(disc)
            del(self.vecAlu[ind])
    
    def checkInd(self, nome):
        self.vecAlu.index(nome)

    def show(self):
        ss = self.disc + ":"
        for i in range(len(self.vecAlu)):
            ss = ss + "\n" +"\t" + str(self.vecAlu[i])
        return ss

    def getAlu(self):
        return self.vecAlu

File: Trabalho_Final/TF/Final/src/tf/Disciplina.java
package tf;
import java.util.ArrayList;

public class Disciplina {
	String cod = "";
	String id="";
	ArrayList<Integer> VecAlu = new ArrayList<Integer>();
	
	public Disciplina(String id, String cod) {
		this.id = id;
		this.cod = cod;
	}
	
	public String Matricular(int mat){
		String ss="";
		if(check(mat) == -1) {
			VecAlu.add(mat);
			return ss;
		}
		ss = "\n Aluno já matriculado nessa Disciplina \n";
		return ss;
	}
	
	public String Desmatricular(int mat){
		String ss="";
		int ind = check(mat);
		if(ind != -1) {
			VecAlu.remove(ind);
			return ss;
		}
		ss = "\n Aluno não matriculado \n";
		return ss;
	}
	
	public int check(int mat){
		for(int i = 0; i< this.VecAlu.size(); i++){
			if(mat == VecAlu.get(i))
				return i;			
		}
		return -1;		
	}
	
	public String show(){
		String ss = this.id +", "+ this.cod +": " + "[";
		if (this.VecAlu != null && !this.VecAlu.isEmpty()) {
			for(Integer i: this.VecAlu) {
				ss += " " + i;
			}
		}
		ss += " ]";
		return ss;
	}
	
	public String getId() {
		return this.id;
	}
	
	public String getCod() {
		return this.cod;
	}
	
	public ArrayList<Integer> getAlu(){
		return this.VecAlu;
	}
}
